dev/io.openliberty.microprofile.health.4.0.internal/src/io/openliberty/microprofile/health40/internal/HealthCheck40ServiceImpl.java

```diff
@@ -0,0 +1,594 @@
+            }
+        }
+
+    }
+
+    @Modified
+    protected void modified(ComponentContext context, Map<String, Object> properties) {
+        /*
+         * During server.xml update, never check for server env.
+         */
+
+        /*
+         * If system was not valid for health check, skip on checking configuration update. We've already indicated that this system is
+         * not fit for file-based health checks.
+         */
+        if (isValidSystemForFileHealthCheck) {
+            processUpdateIntervalConfig((String) properties.get(HealthCheckConstants.HEALTH_SERVER_CONFIG_FILE_UPDATE_INTERVAL));
+        }
+
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void startFileHealthCheckProcesses() {
+
+        /*
+         * Last flag in the if is the beta guard
+         */
+        if (isValidSystemForFileHealthCheck && isFileHealthCheckingEnabled() && ProductInfo.getBetaEdition()) {
+
+            File startFile = HealthFileUtils.getStartFile();
+            File readyFile = HealthFileUtils.getReadyFile();
+            File liveFile = HealthFileUtils.getLiveFile();
+
+            /*
+             * Start health check process.
+             * First check if start file exists, if it does then we must be updating the fileUpdateInterval during runtime after a started file
+             * has been created. In no way should this exist from a previous run of this server (i.e. crash) as validation of the system would have
+             * failed and this we would never get here if that was the case.
+             *
+             *
+             * Perform a check immediately and if status is DOWN, start the process.
+             */
+            if (!startFile.exists() && performFileHealthCheck(startFile, HealthCheckConstants.HEALTH_CHECK_START).equals(Status.DOWN)) {
+                startedTimer = new Timer(false);
+                startedTimer.schedule(new FileUpdateProcess(startFile, HealthCheckConstants.HEALTH_CHECK_START, true), 0, 1000);
+            }
+
+            /*
+             * Perform an immediate check and then start the processes for checking ready and live status.
+             */
+            performFileHealthCheck(readyFile, HealthCheckConstants.HEALTH_CHECK_READY);
+            readyTimer = new Timer(false);
+            readyTimer.schedule(new FileUpdateProcess(readyFile, HealthCheckConstants.HEALTH_CHECK_READY), 0, fileUpdateIntevalMilliseconds);
+
+            performFileHealthCheck(liveFile, HealthCheckConstants.HEALTH_CHECK_LIVE);
+            liveTimer = new Timer(false);
+            liveTimer.schedule(new FileUpdateProcess(liveFile, HealthCheckConstants.HEALTH_CHECK_LIVE), 0, fileUpdateIntevalMilliseconds);
+
+        }
+
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext cc, int reason) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "HealthCheckServiceImpl is deactivated");
+        }
+
```

Comment:
Should we call `stopTimers()` here too, just to be safe?

Reply:
Sure we can.

Reply:
fixed with https://github.com/OpenLiberty/open-liberty/pull/30811/commits/96f6b7f3a6322a9f5290bc471b75ec0c4028c4dc